test/hotspot/jtreg/runtime/cds/appcds/WrongClasspath.java

```diff
@@ -40,14 +40,29 @@ public class WrongClasspath {
 
   public static void main(String[] args) throws Exception {
     String appJar = JarBuilder.getOrCreateHelloJar();
+    String appJarInWorkDir = JarBuilder.getOrCreateHelloJarInWorkDir();
     String unableToUseMsg = "Unable to use shared archive";
     String mismatchMsg = "shared class paths mismatch";
     String hintMsg = "(hint: enable -Xlog:class+path=info to diagnose the failure)";
 
+    // Dump CDS archive with hello.jar
+    // Run with a jar file that differs from the original jar file by the first character only: -cp mello.jar
+    // Shared class paths mismatch should be detected.
+    String mellojar = appJarInWorkDir.replace("hello.jar", "mello.jar");
```

Comment:
I am not sure if we need to add more APIs to JarBuilder. Since you just need two files "hello.jar" and "mello.jar", I think that can be done like:

```
import java.nio.file.Paths;
...
String helloJar = "hello.jar";
String melloJar = "mello.jar";
Files.copy(Paths.get(appJar), Paths.get(helloJar), StandardCopyOption.COPY_ATTRIBUTES);
Files.copy(Paths.get(appJar), Paths.get(melloJar), StandardCopyOption.COPY_ATTRIBUTES);
```

Reply:
yup, that would do as well. Pushed a commit as per the suggestion.